Framework/LiveData/src/LoadLiveData.cpp

```diff
@@ -391,6 +391,41 @@ Workspace_sptr LoadLiveData::appendMatrixWSChunk(Workspace_sptr accumWS,
   return accumWS;
 }
 
+//----------------------------------------------------------------------------------------------
+/** Resets all HistogramX in given EventWorkspace(s) to a single bin.
+ *
+ * Ensures bin boundaries encompass all events currently in the workspace.
+ * This will overwrite any rebinning that was previously done.
+ *
+ * Input should be an EventWorkspace or WorkspaceGroup containing
+ * EventWorkspaces. Any other workspace types are ignored.
+ *
+ * @param workspace :: Workspace(Group) that will have its bins reset
+ */
+void LoadLiveData::resetAllXToSingleBin(API::Workspace &workspace) {
```

Comment:
I think we are dealing with minor points here, that I definitely wouldn't want to see delaying things. However, as per your comments on slack... since the `shared_ptr` is nullable, `API::Workspace * const workspace` might be a better signature for the input, and allow you to avoid the try-catch provided below.